tests/unit/test_irc_bridge.py

```diff
@@ -0,0 +1,238 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""IRC bridge unit tests."""
+
+# pylint: disable=protected-access
+
+from unittest import mock
+from unittest.mock import ANY, MagicMock
+
+import ops
+import pytest
+import yaml
+from ops.testing import Harness
+
+import synapse
+import synapse.workload
+from irc_bridge import IRCBridge
+from synapse.workload import (
+    IRC_BRIDGE_CONFIG_PATH,
+    IRC_BRIDGE_REGISTRATION_PATH,
+    CreateIRCBridgeConfigError,
+    CreateIRCBridgeRegistrationError,
+    WorkloadError,
+    _get_irc_bridge_app_registration,
+    create_irc_bridge_app_registration,
+    create_irc_bridge_config,
+)
+
+
+def test_on_collect_status_service_exists(
+    harness: Harness, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    """
+    arrange: start the Synapse charm, set server_name, mock get_services to return something.
+    act: call _on_collect_status.
+    assert: no actions is taken because the service exists.
+    """
+    harness.update_config({"enable_irc_bridge": True})
+    harness.begin_with_initial_hooks()
+    harness.set_leader(True)
+    container: ops.Container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+    monkeypatch.setattr(container, "get_services", MagicMock(return_value=MagicMock()))
+    enable_irc_bridge_mock = MagicMock()
+    monkeypatch.setattr(IRCBridge, "enable_irc_bridge", enable_irc_bridge_mock)
+
+    event_mock = MagicMock()
+    harness.charm._irc_bridge._on_collect_status(event_mock)
```

Comment:
There is a `harness.evaluate_status` function, maybe that is more realistic than directly calling the event handler?